### PR TITLE
Mean Docstring

### PR DESCRIPTION
**Pull request by MatthewMec**
I am trying to add a second pull request for this. I apologize if I did it wrong.

### CODE REVIEW

**Comment by MarcoGorelli** [comment]
thanks - can you do
```
git fetch upstream
git rebase upstream/main
git push origin HEAD --force
```
? you have lots of changes here from your previous PR

**Comment by MatthewMec** [comment]
Yes sorry I was seeing that was trying to fix, thank you for the fix.

File: narwhals/expression.py [hunk 1]
```diff
@@ -2375,13 +2375,58 @@ def sum(*columns: str) -> Expr:
...
 
 
 def mean(*columns: str) -> Expr:
     """
-    Instantiate an expression representing the mean of one or more columns, similar to `polars.mean`.
+    Get the mean value
+
+
+    Note:
+        Syntactic sugar for ``nw.col('name').mean()``
```
Comment (MarcoGorelli) [inline narwhals/expression.py:2388]:
```suggestion
        Syntactic sugar for ``nw.col(columns).mean()``
```

File: narwhals/expression.py [hunk 2]
```diff
@@ -2375,13 +2375,58 @@ def sum(*columns: str) -> Expr:
...
+
+    Examples:
+        >>> import pandas as pd
+        >>> import polars as pl
+        >>> import narwhals as nw
+        >>> df_pl = pl.DataFrame({"a": [1, 8, 3]})
+        >>> df_pd = pd.DataFrame({"a": [1, 8, 3]})
+
+
+    We define a dataframe agnostic function:
```
Comment (MarcoGorelli) [inline narwhals/expression.py:2404]:
```suggestion
        We define a dataframe agnostic function:
```

File: narwhals/expression.py [hunk 3]
```diff
@@ -2375,13 +2375,58 @@ def sum(*columns: str) -> Expr:
...
+
+
+        >>> def func(df_any):
+        ...     df = nw.from_native(df_any)
+        ...     df = df.select(nw.mean('a'))
+        ...     return nw.to_native(df)
+
+
+    We can then pass either pandas or Polars to `func`:
```
Comment (MarcoGorelli) [inline narwhals/expression.py:2413]:
```suggestion
        We can then pass either pandas or Polars to `func`:
```

File: narwhals/expression.py [hunk 4]
```diff
@@ -2375,13 +2375,58 @@ def sum(*columns: str) -> Expr:
...
+
+
+        >>> func(df_pd)
```
Comment (MarcoGorelli) [inline narwhals/expression.py:2416]: there's a lot of extra whitespace here - could we just have a single newline where necessary please?

**Pull request closed without merging**